kubernetes/src/kubernetes/config/incluster_config.rb

```diff
@@ -62,19 +65,42 @@ def token_file
       @token_file
     end
 
+    def token_refresh_period
+      @token_refresh_period ||= TOKEN_REFRESH_PERIOD
+      @token_refresh_period
+    end
+
     def load_token
       File.open(token_file) do |io|
         self.token = io.read.chomp
+        self.token_expires_at = Time.now + token_refresh_period
```

Comment:
Naming: iiuc, the token has an actual validity/expiration time, that's probably not 1 minute, right?  If so, this name may be misleading.
Maybe something like `token_reload_at` or `token_cached_until`...

Reply:
Or maybe use a hidden instance var to not worry about public name :wink:

Reply:
for the clients, they don't need to see the actual token expiration. it's kinda intentional to have a shorter expiration time exposed to client.